src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -33,6 +32,8 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
 
                     builder.AddCommandLine(new[] { "--urls", ConfigurationHelper.JoinValue(settings.Urls ?? Array.Empty<string>()) });
                 })
+                .ConfigureDefaults(args: null)
```

Comment:
So you want all the values specified in the HostBuilderSettings to be replaced by the defaults, (such as those from appsettings.json and enviornment variables). Then we process our own custom json files + key value pairs, followed by our environment variables. Consider updating the comments in ConfigureKestrel to reflect this behavior.

Reply:
> So you want all the values specified in the HostBuilderSettings to be replaced by the defaults, (such as those from appsettings.json and enviornment variables). Then we process our own custom json files + key value pairs, followed by our environment variables. Consider updating the comments in ConfigureKestrel to reflect this behavior.

I think it would be inconsistent to make the urls argument overridable by everything, but not the metrics arguments and the diagnostic port arguments. In general, I went with the approach that all of the values from the command line (notable exception is the temp API key generation) are considered to be at the same configuration ordering.